src/fmt.jl

```diff
@@ -29,7 +29,10 @@ default_spec!(::Type{T}, ::Type{K}) where {T,K} =
     (DEFAULT_FORMATTERS[T] = DEFAULT_FORMATTERS[K]; nothing)
 
 # seed it with some basic default formatters
-for (t, c) in [(Integer,'d'), (AbstractFloat,'f'), (AbstractChar,'c'), (AbstractString,'s')]
+ComplexInteger = Complex{T} where T<:Integer
```

Comment:
I wanted automatic `Integer` formatting for all `Complex{Integer}` types in one go, that's why I changed the dict type tp Union{DataType,UnionAll}.
So I can write `fmt_default!(Complex{T} where T<:Integer}` and also `default_spec!(Complex{T} where T<:Integer}, 'f')`
Otherwise we would need to to this for all subtypes individually.
The same hold for any other package that has a similar type system without an abstract supertype.